pages/glossary/programatically-determined-link-context.md

```diff
@@ -0,0 +1,19 @@
+--
+title: Programmatically Determined Link Context
+key: programmatically-determined-link-context
+--
+
+In HTML, information that is programmatically determinable from a link includes [text nodes][] that are [included in the accessibility tree][] with a non-empty ("") [accessible name][] and are
+
+- sibling nodes in the same paragraph (HTML `p` element) as the link;
+- sibling nodes in the same list (HTML element with the semantic role of `listitem`) as the link;
+- sibling nodes in the same table cell (HTML element with the semantic role of `cell` or `gridcell`) as the link;
```

Comment:
I don't think sibling nodes are sufficient here. For example:

```html
<p>Go to the <span class="coolStypes">Wikipedia</span> <a href="http://wikipedia.org/">homepage</a></p>
```

In this case the text node "Wikipedia" is not a sibling node of the link. Or alternatively you could have something like this:

```html
<p>Go to the Wikipedia <u><a href="http://wikipedia.org/">homepage</u></a></p>
```

Since the link is wrapped in `u`, the link has no sibling nodes.